File: src/main/java/com/piveguyz/empickbackend/orgstructure/member/command/application/dto/MemberSignUpRequestDTO.java
package com.piveguyz.empickbackend.orgstructure.member.command.application.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
public class MemberSignUpRequestDTO {

    private String password;

    @NotBlank
    private String name;

    private String birth;

    @NotBlank
    private String phone;

    @NotBlank
    private String pictureUrl;

    @NotBlank
    private String email;

    @NotBlank
    private String address;

    private Integer vacationCount = 0;

    private LocalDateTime hireAt;

    private LocalDateTime resignAt;

    private Integer deletedMemberId;
    private Integer updatedMemberId;
    private LocalDateTime lastLoginAt;

    @NotNull
    private Integer status = 1;

    private Integer departmentId;
    private Integer positionId;
    private Integer jobId;
    private Integer rankId;
}

File: Empick-FE/src/dto/member/memberSignUpRequestDTO.js
export default class MemberSignUpRequestDTO {
    constructor({
        password = '',
        name = '',
        birth = '',
        phone = '',
        // pictureUrl 제거 - 백엔드에서 UUID로 자동 생성
        email = '',
        address = '',
        vacationCount = 0,
        hireAt = '',
        resignAt = '',
        deletedMemberId = 0,
        updatedMemberId = 0,
        lastLoginAt = '',
        status = 1, // 1: 활성, 0: 비활성 (기본값 수정)
        departmentId = null,
        positionId = null,
        jobId = null,
        rankId = null,
    } = {}) {
        this.password = password;
        this.name = name;
        this.birth = birth;
        this.phone = phone;
        // this.pictureUrl 제거 - 백엔드에서 UUID로 자동 생성됨
        this.email = email;
        this.address = address;
        this.vacationCount = vacationCount;
        this.hireAt = hireAt;
        this.resignAt = resignAt;
        this.deletedMemberId = deletedMemberId;
        this.updatedMemberId = updatedMemberId;
        this.lastLoginAt = lastLoginAt;
        this.status = status;
        this.departmentId = departmentId;
        this.positionId = positionId;
        this.jobId = jobId;
        this.rankId = rankId;
    }

    /**
     * FormData로 변환 (프로필 이미지 제외)
     * @returns {Object} 폼 데이터 객체
     */
    toFormData() {
        const data = {};

        // null이 아닌 값들만 추가
        Object.keys(this).forEach(key => {
            if (this[key] !== null && this[key] !== undefined && this[key] !== '') {
                data[key] = this[key];
            }
        });

        return data;
    }
}